text/0524-vault-standard.md

```diff
@@ -0,0 +1,949 @@
+  - **Input**: None
+  - **Output**:
+
+    | Field                 | Type    | Description                 |
+    | --------------------- | ------- | --------------------------- |
+    | `jettonDepositGasFee` | `Coins` | Gas fee for Jetton deposit. |
+
+- **`previewExtraCurrencyDepositFee`**
+
+  - **Description**: Returns the gas fee required for depositing Extra Currency to the vault. This fee is charged by the vault contract to the `initiator`
+  - **Requirements**:
+    - MUST exclude forward and storage fees.
+    - MUST exclude fees for subsequent contract interactions (e.g., Jetton Transfer).
+    - MUST NOT revert.
+    - SHOULD return a conservative estimate.
+  - **Input**: None
+  - **Output**:
+
+    | Field                        | Type    | Description                         |
+    | ---------------------------- | ------- | ----------------------------------- |
+    | `extraCurrencyDepositGasFee` | `Coins` | Gas fee for Extra Currency deposit. |
+
+- **`previewWithdrawFee`**
+
+  - **Description**: Returns the gas fee required for withdrawing from the vault. This fee is charged by the vault contract to the `initiator`
+  - **Requirements**:
+    - MUST exclude forward and storage fees.
+    - MUST exclude fees for subsequent contract interactions (e.g., Jetton Transfer).
+    - MUST NOT revert.
+    - SHOULD return a conservative estimate.
+  - **Input**: None
+  - **Output**:
+
+    | Field            | Type    | Description             |
+    | ---------------- | ------- | ----------------------- |
+    | `withdrawGasFee` | `Coins` | Gas fee for withdrawal. |
+
+- **`previewProvideQuoteFee`**
+
+  - **Description**: Returns the gas fee required for querying `totalAssets` and `totalSupply`. This fee is charged by the vault contract to the `initiator`
+  - **Requirements**:
+    - MUST exclude forward and storage fees.
+    - MUST exclude fees for subsequent contract interactions (e.g., Jetton Transfer).
+    - MUST NOT revert.
+    - SHOULD return a conservative estimate.
+  - **Input**: None
+  - **Output**:
+
+    | Field             | Type    | Description              |
+    | ----------------- | ------- | ------------------------ |
+    | `provideQuoteFee` | `Coins` | Gas fee for quote query. |
+
+### Events
+
+- **`Deposited`**
+
+  - **Description**: Emitted when initiator exchanges `depositAmount` for shares, transferring them to receiver.
+  - **Message**:
+
+  <a id="topic_deposited"></a>
+  | Field | Type | Description |
+  |---------------------|-----------------------------|-------------|
+  | `TOPIC_DEPOSITED` | [Opcode](#opcode) | `0x11475d67` |
+  | `initiator` | `Address` | Address initiating the deposit. |
+  | `receiver` | `Address` | Address receiving shares. |
+  | `depositAsset` | Cell<[Asset](#asset)>? | Deposited asset (required for multi-asset vaults, null for single asset). |
+  | `depositAmount` | `Coins` | Deposited asset amount. |
+  | `shares` | `Coins` | Minted shares. |
```

Comment:
Since underlying amount is subject to change it would be great to add `totalAssetAmount` in the log event. It will allow to reconstruct the vault TVL simply by accessing all the logs messages without dealing with get methods invocations.